graphsync.go

```diff
@@ -5,18 +5,44 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/google/uuid"
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-ipld-prime"
 	"github.com/ipld/go-ipld-prime/traversal"
 	"github.com/libp2p/go-libp2p-core/peer"
 )
 
 // RequestID is a unique identifier for a GraphSync request.
-type RequestID int32
```

Comment:
@mvdan first major problem I'm having getting the bindnode work working with uuid RequestIDs is that I'm using this pattern to represent RequestIDs, but they need to be encoded as `Bytes` in dag-cbor. So Im getting this error out of bindnode:

```
panic: bindnode: schema type GraphSyncRequestID is not compatible with Go type graphsync.RequestID: kind mismatch; need slice of bytes
```

So .. we either need to do a translation for this (there's a `ParseRequestID([]byte)` below but it has an error return) or something more fancy for bindnode to do this translation.

Reply:
Unless this int32-vs-bytes conversion is supported as a schema representation strategy, bindnode doesn't support it right now - it wants the types to be equivalent without further action. The conversion should probably just become part of the "to IPLD" and "from IPLD" funcs.

Reply:
I believe the IPLD version of the protocol should ONLY use UUIDs. The representation should just be bytes.

Reply:
The solution I believe is to change the schema -- this is after all the new protocol.